integration_tests/suite/base/test_context_range.py

```diff
@@ -0,0 +1,290 @@
+        {'start': '8005551000', 'end': '8005551009'},
+    ],
+)
+def test_search_all_types(internal_context, incoming_context):
+    response = confd.contexts(internal_context['id']).ranges('user').get()
+    assert_that(
+        response.json,
+        has_entries(
+            total=3,
+            items=contains_inanyorder(
+                {'start': '1000', 'end': '1099'},
+                {'start': '1200', 'end': '1499'},
+                {'start': '1800', 'end': '1899'},
+            ),
+        ),
+    )
+
+    response = confd.contexts(internal_context['id']).ranges('group').get()
+    assert_that(
+        response.json,
+        has_entries(
+            total=1,
+            items=contains_inanyorder(
+                {'start': '2000', 'end': '2999'},
+            ),
+        ),
+    )
+
+    response = confd.contexts(internal_context['id']).ranges('queue').get()
+    assert_that(
+        response.json,
+        has_entries(
+            total=1,
+            items=contains_inanyorder(
+                {'start': '3000', 'end': '3999'},
+            ),
+        ),
+    )
+
+    response = confd.contexts(internal_context['id']).ranges('conference').get()
+    assert_that(
+        response.json,
+        has_entries(
+            total=1,
+            items=contains_inanyorder(
+                {'start': '4000', 'end': '4999'},
+            ),
+        ),
+    )
+
+    response = confd.contexts(incoming_context['id']).ranges('incall').get()
+    assert_that(
+        response.json,
+        has_entries(
+            total=1,
+            items=contains_inanyorder(
+                {'start': '8005551000', 'end': '8005551009'},
+            ),
+        ),
+    )
+
+
+@fixtures.context(
+    user_ranges=[
+        {'start': '1000', 'end': '1999'},
+    ],
+)
+@fixtures.user()
+@fixtures.line_sip()
+def test_search_available(context, user, line):
+    @fixtures.extension(context=context['name'], exten='1005')
+    def aux(extension):
+        with a.user_line(user, line), a.line_extension(line, extension):
+            response = confd.contexts(context['id']).ranges('user').get(available=True)
+            assert_that(
+                response.json,
+                has_entries(
+                    total=2,
+                    items=contains_inanyorder(
+                        {'start': '1000', 'end': '1004'},
+                        {'start': '1006', 'end': '1999'},
+                    ),
+                ),
+            )
+
+            response = confd.contexts(context['id']).ranges('user').get(available=False)
+            assert_that(
+                response.json,
+                has_entries(
+                    total=1,
+                    items=contains_inanyorder(
+                        {'start': '1000', 'end': '1999'},
+                    ),
+                ),
+            )
+
+    aux()
+
+
+@fixtures.context(
+    user_ranges=[
+        {'start': '00', 'end': '99'},
+        {'start': '0001', 'end': '0499'},
```

Comment:
Open question: can you confirm we cannot have this case (ranges that conflict each other) :
For example:  
```
{'start': '00', 'end': '99'}
and {'start': '70', 'end': '199'}
```
?

Reply:
Good catch, I'll have to check to be sure that the good thing happens in this case.
Your example does not work since start and end needs to have the same length but
`{'start': '000', 'end': '099'}, {'start': '70', 'end': '199'}` should result in `{'start': '000', 'end': '199'}`

Reply:
The following scenario fails
```python
@fixtures.context(
    user_ranges=[
        {'start': '100', 'end': '119'},
        {'start': '000', 'end': '099'},
        {'start': '070', 'end': '199'},
    ],
)
def test_search_overlapping_ranges(context):
    response = confd.contexts(context['id']).ranges('user').get()
    print(response)
    assert_that(
        response.json,
        has_entries(
            total=1,
            items=contains_inanyorder(
                {'start': '000', 'end': '199'},
            ),
        ),
    )
 ```
 The result with the current implementation is
 ```javascript
 {'items': [{'end': '119', 'start': '000'}], 'total': 1}
 ```